sanic/response.py

```diff
@@ -73,28 +73,121 @@
 }
 
 
-class HTTPResponse:
+class BaseHTTPResponse:
+    def _encode_body(self, data):
+        try:
+            # Try to encode it regularly
+            return data.encode()
+        except AttributeError:
```

Comment:
Can we check if it's already bytes here? Then we can avoid the round trip from bytes -> str -> bytes in that case.